### PR TITLE
DEV: Refactor sidebar components to prepare for integration into hamburger dropdown

### PR DESCRIPTION
**Pull request by tgxworld**
Minor style change I snuck in

### Before

Peek 2022-07-19 14-042

### After

Peek 2022-07-19 14-04

### CODE REVIEW

File: app/assets/stylesheets/common/base/sidebar.scss [hunk 1]
```diff
@@ -105,285 +105,4 @@
...
-      height: 0.75em;
-    }
-    &.urgent svg {
-      color: $success;
-    }
-    &.unread svg {
-      color: var(--tertiary-med-or-tertiary);
-    }
-  }
-  &.sidebar-section-chat-dms {
```
Comment (tgxworld) [inline app/assets/stylesheets/common/base/sidebar.scss:348]: @lis2 Minor but do note that we do not want to be including chat specific stuff in core.

File: app/assets/javascripts/discourse/app/components/sidebar/sections.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { customSections as sidebarCustomSections } from "discourse/lib/sidebar/custom-sections";
+
+export default class SidebarSections extends GlimmerComponent {
+  customSections;
+
+  constructor() {
+    super(...arguments);
+    this.customSections = this._customSections;
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/sidebar/sections.js:9]: Why is this needed? Can't you just do `get customSections() {}` below and use that directly in the component?

**Review by martin-brennan — APPROVED** [review]
Small question otherwise looks fine since it's just moving stuff around + css

File: app/assets/javascripts/discourse/app/components/sidebar/sections.js (hunk 2 above)
Comment (tgxworld) [reply app/assets/javascripts/discourse/app/components/sidebar/sections.js:9]: If I use a getter, we have to compute the getter each time the getter is called. I'm using a public field here as a form of memoization, since the public field is only assigned once.

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/sidebar/sections.js:9]: I thought you can just use ` @computed` for that with no arguments, no?

Comment (tgxworld) [reply app/assets/javascripts/discourse/app/components/sidebar/sections.js:9]: `@computed` is no longer valid in Glimmer components. Technically, `@computed` with no arguments is also not really valid because it is not considered a computed property when it does not depend on other properties.

**Comment by tgxworld** [comment]
Thank you for reviewing @martin-brennan :+1:
